qiita_pet/static/js/networkVue.js

```diff
@@ -742,79 +705,75 @@ Vue.component('processing-graph', {
       let vm = this;
       var container = document.getElementById('processing-network-div');
       container.innerHTML = "";
-
-      vm.nodes_ds = new vis.DataSet(vm.nodes);
-      vm.edges_ds = new vis.DataSet(vm.edges);
-      var data = {
-        nodes: vm.nodes_ds,
-        edges: vm.edges_ds
+      // Making sure the network is available
+      $("#processing-network-div").show();
+      $("#processing-network-instructions-div").show();
+
+      var layout = {
+        name: 'dagre',
+        rankDir: 'LR',
+        directed: true,
+        nodeDimensionsIncludeLabels: true,
+        nodeSep: 2,
+        spacingFactor: 1.2,
+        padding: 5
       };
-      var options = {
-        autoResize: true,
-        clickToUse: true,
-        nodes: {
-          font: {
-            size: 15,
-            color: '#000000'
-          },
-          size: 13,
-          borderWidth: 2,
-        },
-        edges: {
-          color: 'grey'
-        },
-        layout: {
-          randomSeed: 1,
-          improvedLayout: true,
-          hierarchical: {
-            direction: "LR",
-            sortMethod : 'directed',
-            levelSeparation: 190,
-            shakeTowards: 'roots'
-          }
-        },
-        interaction: {
-          dragNodes: false,
-          dragView: true,
-          zoomView: true,
-          selectConnectedEdges: true,
-          navigationButtons: true,
-          keyboard: false
-        },
-        groups: {
-          jobs: {
-            color: '#FF9152'
-          },
-          artifact: {
-            color: '#FFFFFF'
-          },
-          type: {
-            color: '#BBBBBB'
-          }
-        }
+      var style = [{
+        selector: 'node',
+        style: {
+          'content': 'data(label)',
+          'background-color': 'data(color)',
+          'shape': 'data(shape)',
+          'text-opacity': 0.7,
+          'text-wrap': "wrap",
+          'border-color': '#BBBBBB',
+          'border-width': '1px'
+        }}, {
+        selector: 'edge',
+        style: {
+          'curve-style': 'bezier',
+          'target-arrow-shape': 'triangle'
+        }},
+      ];
+      var panzoom_options =	{
+        zoomOnly: true,
+        sliderHandleIcon: 'fa fa-minus',
+        zoomInIcon: 'fa fa-plus',
+        zoomOutIcon: 'fa fa-minus',
+        resetIcon: 'fa fa-expand'
       };
 
-      vm.network = new vis.Network(container, data, options);
-      vm.network.on("stabilized", function (params) {
+      // Note: we only need to sort the edges to keep the same structure of the
+      //       graph; in other words, nodes order is not important
+      vm.edges = vm.edges.sort((a, b) => (a.data.source > b.data.source) ? 1 : (a.data.source === b.data.source) ? ((a.data.target > b.data.target) ? 1 : -1) : -1 );
```

Comment:
Triple nested ternary is a bit hard... any chance this could be refactored? I think it would be:

```suggestion
      var sort_order = -1;
      if ((a.data.source > b.data.source) || ((a.data.source === b.data.source) && (a.data.target > b.data.target)) {
          sort_order = 1;
      }
      
      vm.edges = vm.edges.sort(sort_order);
```

Reply:
Sounds good, just note that this is a function like in python `list.sort(key=sorting_function)`.

Reply:
oh right, so pack into a function... but main item was, if okay, to expand the nested ternaries :)